### ISSUE TITLE
Plugin request

### ISSUE THREAD

**Issue by ghost** [cuberite/MeetupAgenda2016#1]
Hey. 

I think I won't attend, so I present my ideas here. 
1. Plugin manager / repo / updater
   Preferable with some sort of git, either in lua or c++
   Then a repo containing git links and descriptions (and reading the plugin to seek for commands / permission collisions and search function) 
   At last a webinterface impkemention
2. Better CI
   My imagination: there is a cuberite server running anywhere 24/7. Everytime there is a pr on github, the "master" receives a "note" via webhook, sets a "got it" note on the pr (yellow) and checks the code out and builds it (and tries to run it). After that, he sends github a  "testing required" note (yellow) and waits. Additionally it creates some sort of sign ingame. After that anyone may join the server, eg click the sign and in the background, the "master" starts the slave cuberite server with the individual source he took from the pr (they may share all the same basic ccache) and redirects the client via bungee to that server. Now, the player can test it ingame and report a working state via an ingame commando. If the master receives such a commando, it sends a success note to github (green), maybe we set up the rule to need eg 3 passed tests ingame before marking it as green. For performance reason, the master starts/stops each slave if Noone is on it. Additionally we make the world flatland and limit it to eg 4 chunks.

**Comment by sphinxc0re** [comment]
Could you split this into two issues?

**Comment by madmaxoft** [comment]
I don't think this is a good hackathon candidate. The plugin manager would benefit greatly from having HTTP / REST API in the server, which is not available yet. It's a good candidate for discussion, though, in order to refine what we want from the manager and how we want to implement it.

**Comment by ghost** [comment]
OK, let's move it to general discussion within the forum (you may close it and create a forum thread)
